src/templates/CentralTemplateProvider.ts

```diff
@@ -258,12 +259,26 @@ export class CentralTemplateProvider implements Disposable {
     }
 
     private async getLatestTemplates(context: IActionContext, provider: TemplateProviderBase, latestTemplateVersion: string): Promise<ITemplates | undefined> {
-        if (!this.templateSource || this.templateSource === TemplateSource.Latest || this.templateSource === TemplateSource.Staging) {
-            context.telemetry.properties.templateSource = 'latest';
-            const result: ITemplates = await provider.getLatestTemplates(context, latestTemplateVersion);
-            await provider.cacheTemplateMetadata(latestTemplateVersion);
-            await provider.cacheTemplates(context);
-            return result;
+        try {
+            if (!this.templateSource || this.templateSource === TemplateSource.Latest || this.templateSource === TemplateSource.Staging) {
+                context.telemetry.properties.templateSource = 'latest';
+                const result: ITemplates = await provider.getLatestTemplates(context, latestTemplateVersion);
+                await provider.cacheTemplateMetadata(latestTemplateVersion);
+                await provider.cacheTemplates(context);
+                return result;
+            }
+        } catch (error) {
+            // If we failed to get Java templates for a non-English language, try English
+            if (provider.language === 'Java') {
+                const language = provider.resourcesLanguage || getScriptResourcesLanguage();
+                if (language !== english) {
+                    provider.resourcesLanguage = english;
+                    const result = provider.getLatestTemplates(context, latestTemplateVersion);
+                    const message: string = localize('languageTemplateWarning', 'The templates for the current language are not supported yet. Defaulting to English templates.');
```

Comment:
nit: I think it would be nice to include the `language` in the error message.
```suggestion
                    const message: string = localize('languageTemplateWarning', '{0} templates are currently not supported. Defaulting to English templates.', language);
```

Reply:
Don't hate me, but I think we should actually have this logic in `getScriptResourcesLanguage`. It seems like we could add a property to check to see if the project is Java or not and just default to English if it is.

Reply:
🫠  Haha ok I will move it!